vpc-peering/core/main.tf

```diff
@@ -35,3 +35,22 @@ resource "aws_security_group_rule" "peer-access" {
   security_group_id = each.key
   description       = "vpc peering from vpc: ${var.vpc_name}"
 }
+
+# Note: These next two resources are only run if you need to allow communication between ECS services.
+# Also, vpc peering must be complete before these are run see docs
+resource "aws_route53_vpc_association_authorization" "create-dns-association" {
+  # If source_vpc is not defined as a VAR then this will NOT run.
+  count   = can(var.source_vpc) && var.source_vpc != null && var.accept_remote_dns == null ? 1 : 0
```

Comment:
Not used `can` before. Just coalesce in these sort of situations. Is this effectively the same as 
`coalesce(var.source_vpc, false) && var.accept_remote_dns == null ...` ?

Reply:
I did originally use coalesce but the TF errored if source_vpc is not set.  I will try it again.

Reply:
`can` works here as the result is a boolean.  Coalesce can retuen a false if the var is not set,  but then TF complains that source_vpc_id is a string.